dex/elrond_dex_router/src/factory.rs

```diff
@@ -132,21 +134,82 @@ pub trait FactoryModule {
         }
     }
 
+    #[view(getPair)]
+    fn get_pair(
+        &self,
+        first_token_id: TokenIdentifier,
+        second_token_id: TokenIdentifier,
+    ) -> Address {
```

Comment:
I think it's better to return `Option<Address>` instead of constantly checking for address == Address::zero()

Reply:
done